extension-manifest-v3/src/utils/engines.js

```diff
@@ -217,27 +179,37 @@ function check(response) {
 }
 
 const updateListeners = new Map();
-export function addUpdateListener(name, fn) {
+export function addChangeListener(name, fn) {
   if (!updateListeners.has(name)) {
     updateListeners.set(name, new Set());
   }
 
   updateListeners.get(name).add(fn);
 }
 
+function notifyListeners(name) {
+  const fns = updateListeners.get(name);
+
+  fns?.forEach((fn) => {
+    try {
+      fn();
+    } catch (e) {
+      console.error(
+        `[engines] Error while calling update listener for "${name}"`,
+        e,
+      );
+    }
+  });
+}
+
 const CDN_HOSTNAME = chrome.runtime.getManifest().debug
   ? 'staging-cdn.ghostery.com'
   : 'cdn.ghostery.com';
 
-async function update(name) {
-  if (
-    // Custom and regional engines are created on demand
-    name === CUSTOM_ENGINE ||
-    name === REGIONAL_ENGINE ||
-    // If the IndexedDB is corrupted, and there is no way to load the engine
-    // from storage, we should not try to update it
-    (await loadFromStorage(name)) === null
-  ) {
+export async function update(name) {
+  // If the IndexedDB is corrupted, and there is no way to load the engine
+  // from the storage, we should skip the update.
+  if ((await loadFromStorage(name)) === null) {
```

Comment:
Could be worth logging a warning here that the update is skipped, because storage is not available.